File: textbook-codes/Kap_03_Polyapprox.m
%-- Kap_03_Polyapprox.m
x = linspace(0,5,1000); y = exp(-x);
x1 = linspace(0,5,11); y1 = exp(-x1);
for i=1:4
  p = polyfit(x1,y1,i-1);
  y2 = polyval(p,x);
  subplot(2,2,i); plot(x,y,x1,y1,'o',x,y2);
end

File: textbook-codes/Kap_03_Polyapprox.jl
#-- Kap_03_Polyapprox.jl
using Polynomials
using Plots
#--
x = range(0.0,stop=5.0,length=1000); y = exp.(-x);
x1 = range(0.0,stop=5.0,length=11); y1 = exp.(-x1);
p = plot(layout=(2,2), legend = false)
for i = 1:4
  pp = fit(x1,y1,i-1);  y2 = pp.(x);
  plot!(p,x1,y1, marker=:o, line = :scatter, subplot = i);
  plot!(x,y, subplot = i); plot!(x,y2, subplot = i)
end
display(p)
